rfcs/2022-03-22-motion-rfc.md

```diff
@@ -0,0 +1,612 @@
+
+We can represent this as a bezier function `cubic-bezier(0, 1, 1, 1)` and a bezier curve:
+
+<img alt="rapid-start-curve" src="./images/motion/rapid-start-curve.png" width="200px">
+
+**Ease In - Ease Out Easing:**
+
+If you're familiar with easing in CSS, you must have come across a transition timing property of `ease-in-out`. This is the same as linear but with a slower acceleration at the beginning and a slower deceleration at the end.
+
+<img alt="ease-in-out-easing" src="./images/motion/ease-in-out-easing.gif" width="300px">
+
+We can represent this as a bezier function `cubic-bezier(0.42, 0, 0.58, 1)` and a bezier curve:
+
+<img alt="ease-in-out-curve" src="./images/motion/ease-in-out-curve.png" width="200px">
+
+> You can experiment with different bezier curves to see how they look here: [cubic-bezier.com](https://cubic-bezier.com/)
+
+#### Tokens - Easing
+> Note: *The naming for these tokens is not finalized yet. We will be updating this in the future.*
+
+- We will be storing **Easing tokens** in `blade/src/tokens/global/motion.ts` as a `cubic-bezier()` function.
+
+- After thorough research & experimentation, our design team has created 3 broad **intents** of easing: 
+  - `standard`
+  -  `entrance`
+  -   `exit` 
+- Each of these intents can have easings which we can use in our motion as per our use case:
+  - `effective`
+  - `revealing`
+  - `wary` (only for standard intent)
+  - `attentive` 
+
+- For example, if we want to have an easing on an object's entrance that needs to grab a user's attention, we can the `entrance.attentive` easing.
+
+- Examples of how these easings would look like with relevant use-cases:
+  > All of the following easings are for the `standard` intent
+  - **Effective**:
+  
+    <img alt="effective-easing" src="./images/motion/effective.gif" width="150px">
+  
+  - **Revealing**:
+
+    <img alt="revealing-easing" src="./images/motion/revealing.gif" width="80px">
+  
+  - **Attentive**:
+
+    <img alt="attentive-easing" src="./images/motion/attentive.gif" width="150px">
+  
+  - **Wary**:
+  
+    <img alt="wary-easing" src="./images/motion/wary.gif" width="150px">
+
+Here is the **list of tokens** that we will store in Blade for `Easing`:
+```js
+easing: {
+  standard: {
+    effective: 'cubic-bezier(0.3, 0, 0.2, 1)',
+    revealing: 'cubic-bezier(0.5, 0, 0, 1)',
+    attentive: 'cubic-bezier(0.5, 0, 0.3, 1.5)',
+    wary: 'cubic-bezier(1, 0.5, 0, 0.5)',
+  },
+  entrance: {
+    effective: 'cubic-bezier(0, 0, 0.2, 1)',
+    revealing: 'cubic-bezier(0, 0, 0, 1)',
+    attentive: 'cubic-bezier(0.5, 0, 0.3, 1.5)',
+  },
+  exit: {
+    effective: 'cubic-bezier(0.17, 0, 1, 1)',
+    revealing: 'cubic-bezier(0.5, 0, 1, 1)',
+    attentive: 'cubic-bezier(0.7, 0, 0.5, 1)',
+  },
+}
+```
+
+For **React Native**, we need to store easing as `Easing.bezier` where `Easing` can be imported from react-native's `Animated` or `react-native-reanimated` depending on the library you are using.
```

Comment:
We should update this section with the discussion we did today morning on call about using makeBezier function.

Reply:
That's okay. That's implementation details. Even after adding a helper on top of it, the end state would still be the same. If I add that here, it would just complicate it for readers.